internal/venice-common/src/main/java/com/linkedin/venice/offsets/OffsetRecord.java

```diff
@@ -147,6 +148,33 @@ public synchronized Map<CharSequence, ProducerPartitionState> getProducerPartiti
     return this.partitionState.producerStates;
   }
 
+  public synchronized void setRealtimeTopicProducerState(
+      String kafkaUrl,
+      GUID producerGuid,
+      ProducerPartitionState state) {
+    partitionState.getRealtimeTopicProducerStates().putIfAbsent(kafkaUrl, new VeniceConcurrentHashMap<>());
```

Comment:
`computeIfAbsent` is better than `putIfAbsent` to avoid unnecessary object creation.

Reply:
Good idea. done.